### ISSUE TITLE
Index out of bounds exception when setting values on a Row

### ISSUE THREAD

**Issue by uzzell** [jtablesaw/tablesaw#1184]
When I create a Row from a Table (without specifying a row number) and try to set a value on the Row, I get an error.

**Steps to reproduce:**
```
Column<?> col = DoubleColumn.create("column", 1);
Table table = Table.create(col);
Row row = new Row(table);
row.setDouble("column", 0);
```

**Actual result:**
This produces `java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 128`.  No wonder: the Row's `rowNumber` equals -1, so it's trying to set the `-1`th entry of the column.

If I use a (valid) row index to create the Row, as in `Row row = new Row(table, 0)`, everything is fine.

**Expected result:**
No exception

**Comment by uzzell** [comment]
Now that I've thought about this for a bit, it looks like this behavior reflects the fact that a Row object represents both a set of data (i.e., a row of a Relation) and an Iterator over sets of data.  When one uses this constructor, the object returned acts like a newly instantiated Iterator.  However--and this is the reason for the surprise--it does not act like a row of a Relation.  One must iterate before any data is exposed.

**Comment by lwhite1** [comment]
Yes, it’s really an integrator as creating rows would be very expensive

**Comment by uzzell** [comment]
It would be helpful if the documentation for this constructor indicated that the Row object created wasn't positioned at any index of the Table.
